catalog_full.py

```diff
@@ -0,0 +1,58 @@
+import os
+import urllib
+import jinja2
+import webapp2
+import logging
+import yaml
+
+import models
+import authorizer
+import schemas
+
+JINJA_ENVIRONMENT = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(os.path.dirname(__file__)),
+    extensions=['jinja2.ext.autoescape'],
+    autoescape=True)
+
+class CatalogFull(webapp2.RequestHandler):
+  def SortByName(self, classes):
+    if not classes:
+      return ''
+    return sorted(classes, key=lambda e: e['name'])
+
+  def CoreClasses(self, classes):
+    return [c for c in classes if 'Core' in c['name']]
+
+  def get(self):
+    auth = authorizer.Authorizer(self)
+    if not auth.HasStudentAccess():
+      auth.Redirect()
+      return
```

Comment:
The code on lines 26-30 is great. I think we should factor out any page or serving session related logic out of the authorizer and put that here after line 30.